ocaml/sdk-gen/java/templates/Class.mustache

```diff
@@ -0,0 +1,177 @@
+    }
+
+    {{/is_empty_class}}
+    {{^is_empty_class}}
+    /**
+    * The XenAPI reference (OpaqueRef) to this object.
+    */
+    protected final String ref;
+
+    /**
+    * For internal use only.
+    */
+    {{{class_name}}}(String ref) {
+        this.ref = ref;
+    }
+
+    /**
+    * @return The XenAPI reference (OpaqueRef) to this object.
+    */
+    @JsonValue
+    public String toWireString() {
+        return this.ref;
+    }
+
+    /**
+    * If obj is a {{{class_name}}}, compares XenAPI references for equality.
+    */
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj instanceof {{{class_name}}})
+        {
+            {{{class_name}}} other = ({{{class_name}}}) obj;
+            return other.ref.equals(this.ref);
+        } else
+        {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return ref.hashCode();
+    }
+
+    /**
+     * Represents all the fields in a {{{class_name}}}
+     */
+    public static class Record implements Types.Record {
+        public String toString() {
+            StringWriter writer = new StringWriter();
+            PrintWriter print = new PrintWriter(writer);
+            {{#fields}}
+                print.printf("%1$20s: %2$s\n", "{{{name_camel}}}", this.{{{name_camel}}});
+            {{/fields}}
+            {{#is_event_class}}
+                print.printf("%1$20s: %2$s\n", "snapshot", this.snapshot);
+            {{/is_event_class}}
+            return writer.toString();
+        }
+
+        /**
+        * Convert a {{{class_name}}}.Record to a Map
+        */
+        public Map<String,Object> toMap() {
+            var map = new HashMap<String,Object>();
+            {{#fields}}
+            map.put("{{{name}}}", this.{{{name_camel}}} == null ? {{{default_value}}} : this.{{{name_camel}}});
+            {{/fields}}
+            return map;
+        }
+
+        {{#fields}}
+        /**
+         * {{{description}}}{{#publish_info}}
+         * {{{publish_info}}}{{/publish_info}}
+         */
+        @JsonProperty("{{{name}}}"){{#is_deprecated}}
+        @Deprecated(since = "{{{deprecated_release}}}"){{/is_deprecated}}
+        public {{{type}}} {{{name_camel}}};
+
+        {{/fields}}
+        {{#is_event_class}}
+            /**
+            * The record of the database object that was added, changed or deleted.
+            * The actual type will be VM.Record, VBD.Record, or similar.
+            */
+            public Object snapshot;
+        {{/is_event_class}}
+    }
+
+    {{/is_empty_class}}
+    {{#methods}}
+    /**
+    * {{{description}}}
+    * Minimum allowed role: {{{minimum_allowed_role}}}
+    * {{{publish_info}}}{{#is_deprecated}}
+    * @deprecated since {{{deprecated_release}}}{{/is_deprecated}}
+    *
+    * @param c The connection the call is made on{{#parameters}}
+    * @param {{{name_camel}}} {{{description}}} {{{publish_info}}}{{/parameters}}{{^returns_void}}
```

Comment:
why here start to use {{{ rather than {{ ? Does the JSON data looks like "<b>xx</b>", so you use this?

Reply:
that was to avoid escaping HTML related characters (such as `<` and `>`, which could be present in the name)

See https://github.com/rgrinberg/ocaml-mustache/blob/master/mustache/lib_test/compat/mustache_v200.mli#L47